.github/workflows/continuous_integration.yml

```
@@ -126,3 +95,61 @@ jobs:
     - name: Check docstrings with pydocstyle
       run: |
         pydocstyle --count -e ethicml
+
+  test_with_pytest:
+    needs: [lint_with_pylint, format_with_black_and_isort, type_check_with_mypy, check_docstrings]
+    runs-on: ubuntu-latest
+    env:
+      POETRY_VIRTUALENVS_CREATE: false
+
+    steps:
+      #----------------------------------------------
+      #       check-out repo and set-up python
+      #----------------------------------------------
+      - name: Check out repository
+        uses: actions/checkout@v2
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      #----------------------------------------------
+      #  -----  install & configure poetry  -----
+      #----------------------------------------------
+      - name: Install Poetry
+        uses: snok/install-poetry@v1.2
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+
+      #----------------------------------------------
+      #       load cached venv if cache exists
+      #----------------------------------------------
+      - name: Set up Poetry cache for Python dependencies
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv2-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
```

I only put the 2 here to invalidate a specific cache in palkit. we should probably remove it
```suggestion
          key: venv-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
```